public/js/src/module/encryptor.js

```diff
@@ -25,6 +34,49 @@ function isSupported() {
         new Uint8Array( 8 ).length === 8;
 }
 
+const isEncryptionEnabledSymbol = Symbol( 'isEncryptionEnabled' );
+
+/**
+ * @param {Survey} survey
+ * @return {boolean}
+ */
+function isEncryptionEnabled( survey ) {
+    return Boolean( survey[isEncryptionEnabledSymbol] );
+}
+
+function serializeEncryptedSurvey( survey ) {
+    if ( isEncryptionEnabled( survey ) ) {
+        return Object.assign( {}, survey, {
+            isEncryptionEnabled: true,
+        } );
+    }
+
+    return survey;
+}
+
+function deserializeEncryptedSurvey( survey ) {
+    if ( survey.isEncryptionEnabled ) {
+        let result = Object.assign( {}, survey );
+
+        delete result.isEncryptionEnabled;
+
+        return setEncryptionEnabled( result );
+    }
+
+    return survey;
+}
+
+/**
+ * @param {Survey} survey
+ * @return {Survey}
+ */
+function setEncryptionEnabled( survey ) {
+    return Object.defineProperty( survey, isEncryptionEnabledSymbol, {
+        configurable: false,
+        value: true,
+    } );
+}
+
```

Comment:
This whole set of changes was motivated by reducing/not adding to business logic coupling across concerns. It’s possibly over-engineered in general to hide implementation with `Symbol`, but I think it’s good in general for encapsulation… and especially around anything security related that might be subject to tampering (or even mistakes that have the same outcome as tampering).